libs/services/src/lib/ftu-data/ftu-data.impl.ts

```diff
@@ -286,12 +292,14 @@ export class FtuDataImplService extends FtuDataService {
    * @param iri
    * @returns cell summaries
    */
-  private findCellSummaries<T extends CellSourceItem>(data: Graph<T>, iri: Iri): T[] {
+  private findCellSummaries<T extends CellSourceItem>(data: Graph<T>, iri: Iri, sources: SourceReference[]): T[] {
+    const sourceIds = sources.map((source) => source.id);
     const item = data['@graph'].filter(({ cell_source }) => cell_source === iri);
-    if (item === undefined || item.length == 0) {
+    const itemFilteredBySource = item.filter((source) => sourceIds.includes(source['dataset_id']));
+    if (itemFilteredBySource === undefined || itemFilteredBySource.length == 0) {
```

Comment:
Remove excessive `itemFilteredBySource === undefined` check as it `itemFilteredBySource` cannot possible ever be undefined